linera-core/src/local_node.rs

```diff
@@ -332,4 +311,26 @@ where
         }
         Ok(())
     }
+
+    /// Given a list of chain IDs, returns a map that assigns to each of them the next block
+    /// height, i.e. the lowest block height that we have not processed in the local node yet.
+    ///
+    /// It makes at most `chain_worker_limit` requests to the local node in parallel.
+    pub async fn next_block_heights(
+        &self,
+        chain_ids: impl IntoIterator<Item = &ChainId>,
+        chain_worker_limit: usize,
+    ) -> Result<BTreeMap<ChainId, BlockHeight>, LocalNodeError> {
+        let futures = chain_ids
+            .into_iter()
+            .map(|chain_id| async move {
+                let local_info = self.chain_info(*chain_id).await?;
+                Ok::<_, LocalNodeError>((*chain_id, local_info.next_block_height))
```

Comment:
Yes, maybe that could be improved.
I think in practice the local chain state should usually already be in memory, so that the chain info can be created quickly, but it's still a bit convoluted.